### ISSUE TITLE
Management of NAMESPACE?

### ISSUE THREAD

**Issue by jennybc** [cscheid/rgithub#38]
How do you feel about using `@import`, `@importFrom`, `@export` from roxygen to build the NAMESPACE file? And about NOT importing the entire namespace of httr, Rook, stringr, and rjson?

Asking if you would be receptive to a PR that uses the roxygen approach more fully to manage NAMESPACE and to be more conservative/selective about the imports.

**Comment by cscheid** [comment]
That would be great. These are all welcome changes, and the fact that I didn't do them reflects more on my R ecosystem experience than any conscious choice :)

**Comment by cscheid** [comment]
I'm hoping to clean up most of the issues of rgithub between now and the end of the summer. Is this something that's easy to do? Can you explain what are the issues and possibly help me with it? Thanks!

**Comment by jennybc** [comment]
Yes, my proposed changes to handling the NAMESPACE are quite easy and I'd be happy to help.As in I would make the PR if you invited it. I am basically suggesting this approach:

http://r-pkgs.had.co.nz/namespace.html

TLDR; don't blindly important the entire namespace from any packages (so not for httr, Rook, stringr, and json). Either call those functions like so: `pkg::foo()` or import selected functions by having an `@importFrom pkg foo` directive in a roxygen comment somewhere in the package. The `@importFrom` strategy implies a move towards letting `roxygen2` deal with generating the `NAMESPACE` file, which I find to be a nice strategy. The `devtools` package and RStudio IDE provide nice convenience wrappers for this practice.

**Comment by cscheid** [comment]
Consider your PR invited!

**Comment by cscheid** [comment]
Closed (wontfix) by #70.
